app/timetables/admin.py

```diff
@@ -20,18 +25,24 @@ class AdminsInline(admin.TabularInline):
 
 
 @admin.register(Timetable)
-class TimetableAdmin(DefaultAdmin):
+class TimetableAdmin(CommonReadOnlyFieldsAdmin):
     fields = ('name', 'slug', 'code', 'api_key', 'cycle_length',
-              'current_cycle_day', 'description')
+              'current_cycle_day', 'description', 'date_created', 'date_modified')
     inlines = (AdminsInline,)
 
 
 @admin.register(Dish)
-class DishAdmin(DefaultAdmin):
-    fields = ('name', 'slug', 'description')
+class DishAdmin(CommonReadOnlyFieldsAdmin):
+    fields = ('name', 'slug', 'description', 'date_created', 'date_modified')
+
+
+@admin.register(MenuItem)
+class MenuItemAdmin(admin.ModelAdmin):
+    readonly_fields = ('date_created', 'date_modified')
+    fields = ('timetable', 'cycle_day', 'meal', 'meal_option', 'date_created', 'date_modified')
 
```

Comment:
@andela-kanyanwu: If `CommonReadOnlyFieldsAdmin` can't be reused here and anywhere else that has at least one read-only field in common, I think the class isn't common as the name implies. I would rather explicitly state `readonly_fields` for each Admin class even if many of them share same `readonly_fields`.